r/R/csv.R

```diff
@@ -707,7 +707,7 @@ write_csv_arrow <- function(x,
     x <- Table$create(x)
   }
 
-  assert_that(is_writable_table(x))
+  assert_that(is_writable_table(x) || inherits(x, "FileSystemDataset"))
```

Comment:
Would you like to also support (1) arrow_dplyr_query and (2) RecordBatchReader (which despite the issue title is not currently supported)?

Reply:
I've updated it now to support `arrow_dplyr_query` and the more generic `Dataset` instead of specific subtypes, but I don't think we actually want it to take `RecordBatchReader` objects, as creating them directly is part of the lower-level API. Do you agree?

Reply:
@nealrichardson That make sense?